packages/react-charting/src/components/Legends/Legends.test.tsx

```diff
@@ -0,0 +1,144 @@
+const focusZonePropsInHoverCard = {
+  'aria-label': 'Legend 1 selected',
+};
+
+describe('Legends snapShot testing', () => {
+  it('renders Legends correctly', () => {
+    const component = renderer.create(<Legends legends={legends} />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders allowFocusOnLegends correctly', () => {
+    const component = renderer.create(<Legends legends={legends} allowFocusOnLegends={true} />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders canSelectMultipleLegends correctly', () => {
+    const component = renderer.create(<Legends legends={legends} canSelectMultipleLegends={true} />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders styles correctly', () => {
+    const component = renderer.create(<Legends legends={legends} {...styles} />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
+
+describe('AreaChart - basic props', () => {
+  beforeEach(sharedBeforeEach);
+  afterEach(sharedAfterEach);
+
+  it('Should not mount legends when empty', () => {
+    wrapper = mount(<Legends legends={[]} />);
+    const legend = wrapper.getDOMNode().querySelectorAll('[class^="legendContainer"]');
+    expect(legend!.length).toBe(0);
+  });
+
+  it('Should mount legends when not empty', () => {
+    wrapper = mount(<Legends legends={legends} />);
+    const legend = wrapper.getDOMNode().querySelectorAll('[class^="legendContainer"]');
+    expect(legend).toBeDefined();
+  });
+
+  it('Should render overflowProps correctly', () => {
+    wrapper = mount(<Legends legends={legends} {...overflowProps} />);
+    const overflowProp = wrapper.getDOMNode().querySelectorAll('[class^="ms-Layer"]');
+    expect(overflowProp).toBeDefined();
+  });
+
+  it('Should render data-is-focusable correctly', () => {
+    wrapper = mount(<Legends legends={legends} data-is-focusable={true} />);
+    const dataIsFocusable = wrapper.getDOMNode().querySelector('[class^="legend"]')?.getAttribute('data-is-focusable');
+    expect(dataIsFocusable).toBeTruthy();
+  });
+
+  it('Should render focusZonePropsInHoverCard correctly', () => {
+    wrapper = mount(<Legends legends={legends} {...focusZonePropsInHoverCard} />);
+    const overflowProp = wrapper.getDOMNode().querySelector('[class^="legend"]')?.getAttribute('aria-label');
+    expect(overflowProp).toBe('Legend 1 selected');
+  });
+});
+
+describe('Render calling with respective to props', () => {
+  it('No prop changes', () => {
+    const renderMock = jest.spyOn(LegendsBase.prototype, 'render');
+    const props = {
+      legends,
+    };
+    const component = mount(<Legends {...props} />);
+    component.setProps({ ...props });
+    expect(renderMock).toHaveBeenCalledTimes(2);
+    renderMock.mockRestore();
+  });
+  it('prop changes', () => {
+    const renderMock = jest.spyOn(LegendsBase.prototype, 'render');
```

Comment:
Could you also test 1) legend selected 2) legend overflow

Reply:
Added Test cases for legends selected and legends overflow